internal/peer/node/throttle_grpc.go

```diff
@@ -0,0 +1,69 @@
+/*
+Copyright IBM Corp. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package node
+
+import (
+	"context"
+
+	"github.com/hyperledger/fabric/common/semaphore"
+	"github.com/hyperledger/fabric/core/peer"
+	"google.golang.org/grpc"
+)
+
+var grpcSemaphores = make(map[string]Semaphore)
```

Comment:
Please: No package level state. You have a function that returns it and . one consumer.

Reply:
oh I totally missed this...

Reply:
Need the package level var because it is used by unaryThrottleInterceptor and streamThrottleInterceptor, but it cannot be passed in as a parameter.

Reply:
Figured out a way to pass semaphores as a parameter and removed the package level var.